sympy/functions/elementary/exponential.py

```diff
@@ -1052,7 +1055,7 @@ def mul(d1, d2):
         pterms = {}
 
         for term in Add.make_args(p):
-            co1, e1 = coeff_exp(term, x)
+            co1, e1 = coeff_exp(term, t)
             pterms[e1] = pterms.get(e1, S.Zero) + co1.removeO()
```

Comment:
Where does the order term of `co1` come from? Is it the order term of `p`? What if that is removed from the loop:

    for term in Add.make_args(p.removeO()):

Reply:
Yes it does look like that ! Will make the change.